File: backend/app/schemas/interview.py
"""面试 schema。"""
from pydantic import BaseModel, Field


class InterviewCreateRequest(BaseModel):
    resume_id: int | None = None
    job_id: int | None = None
    record_text: str = Field(..., min_length=1, max_length=20000)


class InterviewEvalResult(BaseModel):
    interview_id: int
    summary: str
    capability_eval: dict
    recommendation: str
    rationale: dict

File: frontend/src/api/interview.ts
import { api } from "./client";

export interface CreateInterviewRequest {
  resume_id?: number;
  job_id?: number;
  record_text: string;
}

export interface InterviewEvalResult {
  interview_id: number;
  summary: string;
  capability_eval: Record<string, string>;
  recommendation: string;
  rationale: Record<string, unknown>;
}

export async function createInterview(req: CreateInterviewRequest): Promise<{ interview_id: number; status: string }> {
  const resp = await api.post("/interviews", req);
  return resp.data;
}

export async function evaluateInterview(interviewId: number): Promise<InterviewEvalResult> {
  const resp = await api.post<InterviewEvalResult>(`/interviews/${interviewId}/evaluate`);
  return resp.data;
}
